test/retry.ts

```diff
@@ -499,3 +499,46 @@ test('respect maximum backoff', async t => {
 
 	await server.close();
 });
+
+test('respect custom retry.delay', async t => {
+	const retryCount = 5;
+	let requestCount = 0;
+
+	const server = await createHttpTestServer();
+	server.get('/', (_request, response) => {
+		requestCount++;
+
+		if (requestCount === retryCount) {
+			response.end(fixture);
+		} else {
+			response.sendStatus(500);
+		}
+	});
+
+	// Register observer that asserts on duration when a measurement is performed
+	const obs = new PerformanceObserver(items => {
+		const measurements = items.getEntries();
+
+		const duration = measurements[0].duration ?? Number.NaN;
+		const expectedDuration = 200 + 300 + 400 + 500;
+
+		t.true(Math.abs(duration - expectedDuration) < allowedOffset, `Duration of ${duration}ms is not close to expected duration ${expectedDuration}ms`);
+
+		obs.disconnect();
+	});
+	obs.observe({entryTypes: ['measure']});
+
+	// Start measuring
+	performance.mark('start');
+	t.is(await ky(server.url, {
+		retry: {
+			limit: retryCount,
+			delay: n => 100 * (n + 1),
+		},
+	}).text(), fixture);
+	performance.mark('end');
+
+	performance.measure('linear', 'start', 'end');
+
+	await server.close();
+});
```

Comment:
I feel like the performance mesurement code could be abstracted a bit so we don't have so much duplication between the tests.